pallets/parachain-staking/src/weights.rs

```diff
@@ -372,95 +424,110 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
 	/// The range of component `y` is `[2, 300]`.
 	fn delegate(x: u32, y: u32, ) -> Weight {
 		// Proof Size summary in bytes:
-		//  Measured:  `2479 + x * (79 ±0) + y * (38 ±0)`
-		//  Estimated: `32988 + x * (405 ±0) + y * (195 ±0)`
-		// Minimum execution time: 94_976_000 picoseconds.
-		Weight::from_parts(83_764_148, 32988)
-			// Standard Error: 1_433
-			.saturating_add(Weight::from_parts(154_670, 0).saturating_mul(x.into()))
-			// Standard Error: 470
-			.saturating_add(Weight::from_parts(48_242, 0).saturating_mul(y.into()))
+		//  Measured:  `2424 + x * (79 ±0) + y * (38 ±0)`
+		//  Estimated: `32553 + x * (405 ±0) + y * (195 ±0)`
+		// Minimum execution time: 107_345_000 picoseconds.
+		Weight::from_parts(96_029_633, 32553)
+			// Standard Error: 20_841
+			.saturating_add(Weight::from_parts(167_244, 0).saturating_mul(x.into()))
+			// Standard Error: 6_812
+			.saturating_add(Weight::from_parts(56_328, 0).saturating_mul(y.into()))
 			.saturating_add(T::DbWeight::get().reads(7_u64))
 			.saturating_add(T::DbWeight::get().writes(7_u64))
 			.saturating_add(Weight::from_parts(0, 405).saturating_mul(x.into()))
 			.saturating_add(Weight::from_parts(0, 195).saturating_mul(y.into()))
 	}
 	/// Storage: ParachainStaking DelegatorState (r:1 w:1)
 	/// Proof Skipped: ParachainStaking DelegatorState (max_values: None, max_size: None, mode: Measured)
-	/// Storage: ParachainStaking DelegationScheduledRequests (r:1 w:1)
+	/// Storage: ParachainStaking DelegationScheduledRequests (r:100 w:100)
 	/// Proof Skipped: ParachainStaking DelegationScheduledRequests (max_values: None, max_size: None, mode: Measured)
-	fn schedule_leave_delegators() -> Weight {
-		// Proof Size summary in bytes:
-		//  Measured:  `107`
-		//  Estimated: `7144`
-		// Minimum execution time: 28_255_000 picoseconds.
-		Weight::from_parts(28_539_000, 7144)
-			.saturating_add(T::DbWeight::get().reads(2_u64))
-			.saturating_add(T::DbWeight::get().writes(2_u64))
+	/// The range of component `x` is `[1, 100]`.
+	fn schedule_leave_delegators_worst(x: u32, ) -> Weight {
+		// Proof Size summary in bytes:
+		//  Measured:  `1463 + x * (14395 ±0)`
+		//  Estimated: `7223 + x * (31269 ±0)`
+		// Minimum execution time: 57_741_000 picoseconds.
+		Weight::from_parts(68_369_621, 7223)
+			// Standard Error: 195_409
+			.saturating_add(Weight::from_parts(28_183_556, 0).saturating_mul(x.into()))
+			.saturating_add(T::DbWeight::get().reads(1_u64))
+			.saturating_add(T::DbWeight::get().reads((1_u64).saturating_mul(x.into())))
+			.saturating_add(T::DbWeight::get().writes(1_u64))
+			.saturating_add(T::DbWeight::get().writes((1_u64).saturating_mul(x.into())))
+			.saturating_add(Weight::from_parts(0, 31269).saturating_mul(x.into()))
```

Comment:
I think `31269.saturating_mul(x)` is going to be problematic here because `x = 100` in our runtimes and `31_269 * 100` is way too close to our block limit.

Reply:
One thing to note is that, we have deprecated these methods in favor of batch schedule revoke, so perhaps it's alright for it to have a high enough weight to incentivize people to move away from it.

Additionally now that I notice it this functionality has already passed 6 months of time and should be marked for removal https://github.com/PureStake/moonbeam/pull/1760